### DISCUSSION TITLE
opt_interactive captions not rendering from Quarto

### DISCUSSION THREAD

**Discussion by SusanMcMillan** [rstudio/gt#1361] [Q&A]
I need table captions that are numbered appropriately in an html file, and users need to sort by columns. When I use the opt_interactive function, Quarto (in Rstudio v. 2023.06.0 Build 421 running on Windows) will not produce either cross references or captions correctly. Is there something else I need to do (.qmd file follows)?

---
title: "Sortable Tables Test"
format: html
editor: source
---

This @tbl-gt_non_interactive reference works as expected, as does the caption. 

```{r}
#| label: tbl-gt_non_interactive
#| tbl-cap: "Airquality - {gt} No Interactive Option"
#| echo: true

library(gt)
gt(head(airquality))

```
This reference to @tbl-gt_interactive does **not** work and the caption is not correctly produced. However, the table is now sortable by column, as desired.

```{r}
#| label: tbl-gt_interactive
#| tbl-cap: "Airquality - {gt} Interactive Option"
#| echo: true

gt(airquality)%>%
  opt_interactive()

```

**Comment by olivroy** [comment]
The example you shared works as expected with Quarto > 1.4. You may want to avoid using `_` in your labels to avoid problems with other outputs like pdf.


````qmd
---
title: "Sortable Tables Test"
format: html
editor: source
---


This @tbl-gt_non_interactive reference works as expected, as does the caption.

```{r}
#| label: tbl-gt_non_interactive
#| tbl-cap: "Airquality - {gt} No Interactive Option"
#| echo: true

library(gt)
gt(head(airquality))

```

This reference to @tbl-gt_interactive does not work and the caption is not correctly produced. However, the table is now sortable by column, as desired.

```{r}
#| label: tbl-gt_interactive
#| tbl-cap: "Airquality - {gt} Interactive Option"
#| echo: true

gt(airquality)%>%
  opt_interactive()
```
````

![image](https://github.com/rstudio/gt/assets/52606734/9f190706-3b46-4d98-90ba-601a83a38c85)

https://github.com/quarto-dev/quarto-cli/issues/1084

**Reply by olivroy** [reply]
I edited my answer to say that the example you shared works as expected with Quarto > 1.4
